project/components/resource_request/models.py

```diff
@@ -33,9 +34,21 @@ class ResourceRequest(DBModel):
 
     id = Column(UUID(as_uuid=True), primary_key=True, default=uuid4)
     project_id = Column(UUID(as_uuid=True), ForeignKey('projects.id'), nullable=False)
-    requested_by_user_id = Column(VARCHAR(length=256), nullable=False)
+    user_id = Column(VARCHAR(length=256), nullable=False)
+    username = Column(VARCHAR(length=256), nullable=False)
+    email = Column(VARCHAR(length=256), nullable=False)
     requested_for = Column(VARCHAR(length=256), nullable=False)
     requested_at = Column(TIMESTAMP(timezone=True), default=datetime.utcnow, nullable=False)
     completed_at = Column(TIMESTAMP(timezone=True), nullable=True)
 
     project = relationship('Project', back_populates='resource_requests')
+
+    __table_args__ = (
+        Index(
+            'user_id',
+            'project_id',
+            'requested_for',
+            unique=True,
+        ),
+        {'schema': 'project'},
+    )
```

Comment:
Please move this after `__tablename__`, so that private vars were in one place.
You can also omit  `{'schema': 'project'},` because it's defined upon `DBModel` creation here
https://github.com/PilotDataPlatform/project/blob/bdeb452c95d3b1dcfa2556dfe4e20b61e0207e56/project/components/db_model.py#L19

```suggestion
    __table_args__ = (Index('user_id', 'project_id', 'requested_for', unique=True),)
```